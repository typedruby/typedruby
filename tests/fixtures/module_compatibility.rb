# @typedruby

class MySet::[T]
  def @elements : { T => nil }

  def initialize(Enumerable::[T] elements) => nil
    @elements = elements.map { |x| [x, nil] }.to_h
    nil
  end

  def concat(Enumerable::[T] other) => nil
    other.each do |x|
      @elements[x] = nil
    end
    nil
  end

  def elements => [T]
    @elements.keys
  end
end

def main => nil
  set = MySet.new([1, 2, 3])

  set.concat(["foo", "bar"])

  set.concat({ 4 => 5 })

  set
end
