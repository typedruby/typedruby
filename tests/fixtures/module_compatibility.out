error: Could not match types:

        @ __ROOT__/tests/fixtures/module_compatibility.rb:24
     24 |    set = MySet.new([1, 2, 3])
                              ^ Integer, with:
        @ __ROOT__/tests/fixtures/module_compatibility.rb:26
     26 |    set.concat(["foo", "bar"])
                         ^^^^^ String

        - arising from an attempt to match:

        @ __ROOT__/tests/fixtures/module_compatibility.rb:11
     11 |    def concat(Enumerable::[T] other) => nil
                        ^^^^^^^^^^^^^^^ Enumerable::[Integer], with:
        @ __ROOT__/tests/fixtures/module_compatibility.rb:26
     26 |    set.concat(["foo", "bar"])
                        ^^^^^^^^^^^^^^ Array::[String]
        @ __ROOT__/tests/fixtures/module_compatibility.rb:26
     26 |    set.concat(["foo", "bar"])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^ in this expression

error: Could not match types:

        @ __ROOT__/tests/fixtures/module_compatibility.rb:24
     24 |    set = MySet.new([1, 2, 3])
                              ^ Integer, with:
        @ (builtin stdlib):850
    850 |    include Enumerable::[[KeyType, ValueType]]
                                  ^^^^^^^^^^^^^^^^^^^^ Array::[Integer]

        - arising from an attempt to match:

        @ __ROOT__/tests/fixtures/module_compatibility.rb:11
     11 |    def concat(Enumerable::[T] other) => nil
                        ^^^^^^^^^^^^^^^ Enumerable::[Integer], with:
        @ __ROOT__/tests/fixtures/module_compatibility.rb:28
     28 |    set.concat({ 4 => 5 })
                        ^^^^^^^^^^ Hash::[Integer, Integer]
        @ __ROOT__/tests/fixtures/module_compatibility.rb:28
     28 |    set.concat({ 4 => 5 })
             ^^^^^^^^^^^^^^^^^^^^^^ in this expression

error: Could not match types:

        @ __ROOT__/tests/fixtures/module_compatibility.rb:23
     23 |  def main => nil
                       ^^^ NilClass, with:
        @ __ROOT__/tests/fixtures/module_compatibility.rb:24
     24 |    set = MySet.new([1, 2, 3])
                         ^^^ MySet::[Integer]

        - arising from an attempt to match:

        @ __ROOT__/tests/fixtures/module_compatibility.rb:23
     23 |  def main => nil
                       ^^^ NilClass, with:
        @ __ROOT__/tests/fixtures/module_compatibility.rb:30
     30 |    set
             ^^^ MySet::[Integer]
